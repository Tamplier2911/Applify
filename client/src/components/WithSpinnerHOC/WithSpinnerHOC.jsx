// import "./WithSpinnerHOC.scss";
import React from "react";
import PropTypes from "prop-types";

// JS Rendering CSS
import {
  WithSpinnerHOCContainer,
  WithSpinnerHOCSquare,
} from "./WithSpinnerHOCStyles";

const WithSpinnerHOC = (WrappedComponent) => ({ isLoading, ...otherProps }) => {
  return isLoading ? (
    <WithSpinnerHOCContainer data-test="content-is-loading">
      <WithSpinnerHOCSquare />
    </WithSpinnerHOCContainer>
  ) : (
    <WrappedComponent data-test="content-is-loaded" {...otherProps} />
  );
};

WithSpinnerHOC.propTypes = {
  isLoading: PropTypes.bool,
  otherProps: PropTypes.object,
};

export default WithSpinnerHOC;
