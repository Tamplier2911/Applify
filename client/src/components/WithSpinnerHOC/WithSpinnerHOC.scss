/*

.spinner-container {
  display: grid;
  justify-items: center;
  align-items: center;
  height: 70vh;

  &__square {
    width: 10rem;
    height: 10rem;

    background-color: var(--cl-font);
    animation: rotate 1.5s infinite ease-in-out;
    box-shadow: 0rem 0rem 2rem var(--cl-shadow);
  }
}

@keyframes rotate {
  0% {
    transform: perspective(120px) rotateX(0deg) rotateY(0deg);
  }
  50% {
    transform: perspective(120px) rotateX(-180deg) rotateY(0deg);
  }
  100% {
    transform: perspective(120px) rotateX(-180deg) rotateY(-180deg);
  }
}

*/
