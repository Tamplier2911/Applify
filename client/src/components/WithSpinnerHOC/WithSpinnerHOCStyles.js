import styled, { css } from "styled-components";

const squareAnimation = css`
  @keyframes rotate {
    0% {
      transform: perspective(120px) rotateX(0deg) rotateY(0deg);
    }
    50% {
      transform: perspective(120px) rotateX(-180deg) rotateY(0deg);
    }
    100% {
      transform: perspective(120px) rotateX(-180deg) rotateY(-180deg);
    }
  }
`;

export const WithSpinnerHOCContainer = styled.div`
  display: grid;
  justify-items: center;
  align-items: center;
  height: 70vh;
`;

export const WithSpinnerHOCSquare = styled.div`
  ${squareAnimation}
  width: 10rem;
  height: 10rem;
  background-color: var(--cl-font);
  animation: rotate 1.5s infinite ease-in-out;
  box-shadow: 0rem 0rem 2rem var(--cl-shadow);
`;
