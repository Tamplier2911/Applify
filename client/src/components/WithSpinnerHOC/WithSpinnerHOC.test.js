import React from "react";
import toJson from "enzyme-to-json";

// component
import WithSpinnerHOC from "./WithSpinnerHOC";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<WithSpinnerHOC />", () => {
  let wrapper;
  const TestComponent = () => <div>Hello, World!</div>;
  const TestComponentWithSpinner = WithSpinnerHOC(TestComponent);
  beforeEach(() => {
    wrapper = setShallow(<TestComponentWithSpinner />);
  });

  it("renders without an error", () => {
    const withSpinnerHoc = findByTestAttr(wrapper, "content-is-loaded");
    expect(withSpinnerHoc).toHaveLength(1);
  });

  it("renders loading fallback if isLoading is true", () => {
    wrapper = setShallow(<TestComponentWithSpinner isLoading={true} />);
    const spinnerFallback = findByTestAttr(wrapper, "content-is-loading");
    expect(spinnerFallback).toHaveLength(1);
  });

  it("renders wrapped component if isLoading is false", () => {
    const wrappedComponent = findByTestAttr(wrapper, "content-is-loaded");
    expect(wrappedComponent).toHaveLength(1);
  });

  it("recieves expected props", () => {
    const expectedProps = { isLoading: false, otherProps: {} };
    const err = checkProps(TestComponentWithSpinner, expectedProps, "props");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const withSpinnerHoc = findByTestAttr(wrapper, "content-is-loaded");
    expect(toJson(withSpinnerHoc)).toMatchSnapshot();
  });
});
