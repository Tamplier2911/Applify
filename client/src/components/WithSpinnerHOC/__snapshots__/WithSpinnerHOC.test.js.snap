// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithSpinnerHOC /> matches snapshot 1`] = `
<TestComponent
  data-test="content-is-loaded"
/>
`;
